### PR TITLE
Kinesis Module

### LINKED ISSUE #64029: Springlock MOD module kills you even when you have Memento Mori necklace
## Round ID:
N/A

## Testmerges:
N/A

## Reproduction:
Put a Springlock MOD module inside of MOD control unit, have on your neck slot a Memento Mori that is active. Power your MOD suit and turn on a smoke machine with reagants. Have a horrible death.

### LINKED ISSUE #64136: modsuits cant run out of charge
after the mod core update modsuits use power cell's use() proc instead of overriding the charge themselves.
power cells cant be use()d if the amount of charge required is more than the cell has, so it will never remove the amount of charge required to kill the cell 
im probably gonna fix this by adding a force arg in use() that makes the cell get used regardless

### LINKED ISSUE #64158: trying to install a battery into a modsuit with free storage space places the battery into storage
## Round ID:
[176676](https://scrubby.melonmesa.com/round/176676)

## Testmerges:
N/A

## Reproduction:
open panel on built modsuit, install any storage module, try to install a battery after, get frustrated as you either have to remove your storage module or fill it up to install a battery

### LINKED ISSUE #64161: Modsuit cores disappear in construction
## Round ID:
[176687](https://scrubby.melonmesa.com/round/176687)

## Reproduction:
Construct a modsuit control unit (put in core,screwdriver, add pieces, wrench , screwdriver, armor)
notice how the mod core you put in when constructing disappeared

### LINKED ISSUE #64186: ModSuits : taking out cell doesn't give it back
Reporting client version: 514.1575

## Round ID:
[176766](https://scrubby.melonmesa.com/round/176766)

## Reproduction:

The ModSuit: it was a engineering one that I built in robotics, two cores, bluespace power cells, storage and thermal mods.

After finishing to build it I wanted to remove the cell, took it out, nothing on hand, floor, or storage.
Put a second one, tried to get it out too, disappeared all the same.

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

fixes #64029
fixes #64136
fixes #64158
fixes #64186
fixes #64161
makes mod jetpacks show particles indoors
gives the prototype suit back their upgraded cell (they literally lose 1% of power per 3 seconds with their current one)
Fully reworks the kinesis module to not be TK but lamer, it is now a gravitational anomaly locked module. video here: https://streamable.com/9as50e

## Why It's Good For The Game

Unique stuff that can be used in fun scenarios

## Changelog

:cl:
add: Reworked the MOD Kinesis module and added it to techwebs.
fix: fixed modsuits not running out of charge
fix: fixed modsuits needing multiple cores and cell fitting being weird
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
poor lemon

**Comment by CRITAWAKETS** [comment]
holy shit guys its the gravity gun

**Comment by Pickle-Coding** [comment]
I had fun using it on vending machines.

File: code/modules/clothing/outfits/standard.dm [hunk 1]
```diff
@@ -295,6 +295,15 @@
...
 
+/datum/outfit/centcom/commander/mod
+	name = "CentCom Commander (MODsuit)"
+
+	suit_store = /obj/item/tank/internals/oxygen
+	suit = null
+	head = null
+	mask = /obj/item/clothing/mask/gas/sechailer
+	back = /obj/item/mod/control/pre_equipped/corporate
+
```
Comment (tralezab) [inline code/modules/clothing/outfits/standard.dm:306]: atomize this into its own pr

**Review by tralezab — CHANGES_REQUESTED** [review]
atomization

File: code/modules/mod/modules/module_kinesis.dm [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+		hitting_atom = next_turf
+	for(var/atom/movable/movable_content as anything in next_turf.contents)
+		if(ismob(movable_content))
+			continue
+		if(movable_content.density)
+			hitting_atom = movable_content
+			break
+	var/obj/item/grabbed_item = grabbed_atom
+	grabbed_item.melee_attack_chain(mod.wearer, hitting_atom)
+	COOLDOWN_START(src, hit_cooldown, hit_cooldown_time)
```
Comment (LemonInTheDark) [inline code/modules/mod/modules/module_kinesis.dm:128]: There's a movement loop that does homing, I wonder if you wouldn't be better off using that. Would need to use signals to modify it slightly, but it'd work just fine, and I think would smooth out the snapping movement a bit. would also allow for a higher refresh rate then the process delay.

Making pixel_x/y work would be a bit more tricky, but not too bad

Comment (LemonInTheDark) [reply code/modules/mod/modules/module_kinesis.dm:128]: Ahhh s harder then I thought. This is ok as a bare minimum

**Pull request merged**